lnwallet/interface_test.go

```diff
@@ -2081,6 +2081,63 @@ func testReorgWalletBalance(r *rpctest.Harness, w *lnwallet.LightningWallet,
 	}
 }
 
+// testUnconfirmedTransactionDestAddr tests that when sending an onchain
+// transaction, the destination address is displayed when we call list
+// transactions, even when there have been 0 confirmations.
+func testUnconfirmedTransactionDestAddr(r *rpctest.Harness, alice, bob *lnwallet.LightningWallet,
```

Comment:
Would rather just add this to the existing `testListTransactionDetails` test case.

Reply:
@wpaulino I removed the test entirely, I was able to make an assertion on the number of tx destination addresses returned from a 0 confirmation transaction. It's just under the assertion for 0 confirmations.

```
var mempoolTxFound bool
for _, txDetail := range txDetails {
  if !bytes.Equal(txDetail.Hash[:], burnTXID[:]) {
    continue
  }

  // Now that we've found the transaction, ensure that it has a
  // negative number of confirmations to indicate that it's
  // unconfirmed.
  mempoolTxFound = true
  if txDetail.NumConfirmations != 0 {
    t.Fatalf("num confs incorrect, got %v expected %v", txDetail.NumConfirmations, 0)
  }

  // We test that each txDetail has destination addresses. This ensures
  // that even when we have 0 confirmation transactions, the destination
  // addresses are returned.
  if len(txDetail.DestAddresses) == 0 {
    t.Fatalf("0 conf transactions should display destination addresses")
   }
}
```
I tested on master, and it would not return any destination addresses and fail. On this branch it passes since its updated to return tx destination addresses for 0 confs.

The test is reduced to 2 lines in the existing ```testListTransactionDetails```. 

Let me know if this preferable, I'll also in the future look to follow this style of updating existing tests first.